code/game/turfs/floor_attackby.dm

```diff
@@ -565,6 +565,62 @@
 				else
 					qdel(C)
 				return
+			else if (istype(C, /obj/item/stack/farming/seeds/lime))
+				visible_message("[user] places the seeds in the ploughed field.")
+				new/obj/structure/farming/plant/lime(src)
+				if (C.amount>1)
```

Comment:
So many systems use this same if-else chain for removing stuff, a new global proc should be made that is just something like deleteIfEmpty(src, amounttoremove)